pkg/workloadmeta/collectors/internal/remote/process_collector/process_collector.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package processcollector
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	dderrors "github.com/DataDog/datadog-agent/pkg/errors"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	grpcutil "github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	protoutils "github.com/DataDog/datadog-agent/pkg/util/proto"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta/collectors/internal/remote"
+)
+
+const (
+	collectorID = "process-collector"
+)
+
+type client struct {
+	cl              pbgo.ProcessEntityStreamClient
+	parentCollector *remoteProcessCollectorStreamHandler
+}
+
+func (c *client) StreamEntities(ctx context.Context, opts ...grpc.CallOption) (remote.Stream, error) {
+	log.Debug("[remoteprocesscollector] starting a new stream")
+	streamcl, err := c.cl.StreamEntities(
+		ctx,
+		&pbgo.ProcessStreamEntitiesRequest{},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &stream{cl: streamcl}, nil
+}
+
+type stream struct {
+	cl pbgo.ProcessEntityStream_StreamEntitiesClient
+}
+
+func (s *stream) Recv() (interface{}, error) {
+	log.Trace("[remoteprocesscollector] calling stream recv")
+	return s.cl.Recv()
+}
+
+type remoteProcessCollectorStreamHandler struct {
```

Comment:
You're already in the `processcollector` package, the name is redundant. `streamHandler` is enough.

Reply:
I also removed `[remoteprocesscollector]` from the logs.